### ISSUE TITLE
Change Record statuses to Enum

### ISSUE THREAD

**Issue by dmidlo** [dmidlo/histdata.com-tools#20]

**Comment by dmidlo** [comment]
Reconciled under #179. Partially satisfied: the sidecar lineage through commit 76336c4 introduced WorkStatus and WorkItem.with_status(...) in runtime_contracts.py, and sidecar workflows/activities/status stores use that enum contract. Remaining work: the legacy Record.status field in records.py is still a string compatibility field, so this issue should stay open as the precise final cleanup item for changing or retiring legacy Record.status semantics. #73 was closed as superseded by WorkItem.with_status(...).

**Comment by dmidlo** [comment]
Confirmed again during the #143 follow-up audit after #188: this issue is still open for the right reason and should not be duplicated.

Current state:
- Sidecar contracts use `WorkStatus` and `WorkItem.with_status(...)`.
- Manifest/status stores normalize runtime status through `WorkStatus`.
- Legacy `src/histdatacom/records.py` still exposes `Record.status` as a string compatibility field.

This remains one of the concrete blockers for closing #143 unless the parent scope is narrowed or the legacy `Record` surface is explicitly retired instead of converted.

**Comment by dmidlo** [comment]
Implemented Record status enum normalization in commit 72b7e2f on branch codex/issue-20-record-status-enum. Record.status now normalizes to WorkStatus, preserves unknown legacy status text for migration, serializes legacy metadata via legacy_status, and production Record status writes use WorkStatus members instead of raw strings. Verification: venv/bin/python -m pytest -q (382 passed, 1 skipped); PATH="/Users/davidmidlo/projects/histdata.com-tools/venv/bin:/opt/local/bin:/opt/local/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Library/Apple/usr/bin:/Applications/Little Snitch.app/Contents/Components:/usr/local/share/dotnet:~/.dotnet/tools:/Users/davidmidlo/.codex/tmp/arg0/codex-arg0AWvfyN:/Users/davidmidlo/.antigravity/antigravity/bin:/Users/davidmidlo/.local/bin:/Users/davidmidlo/.nvm/versions/node/v20.18.0/bin:/usr/local/bin/mongodb/bin:/opt/local/bin:/opt/local/sbin:/Users/davidmidlo/bin:/Applications/sqlcl/bin:/usr/local/mysql/bin:/Users/davidmidlo/.rvm/bin:/usr/local/share/dotnet/:/Users/davidmidlo/.dotnet/tools:/Applications/Codex.app/Contents/Resources" venv/bin/pre-commit run --all-files (passed, coverage 80.8%).
